### PR TITLE
feat: New Module `avm/res/network/bastion-host`

### PR DESCRIPTION
**Pull request by krbar**
## Description

New Module `avm/res/network/bastion-host`, migrated from CARML.

## Adding a new module

- [x] A proposal has been submitted and approved.
- [ ] I have included "Closes #{module_proposal_issue_number}" in the PR description.
- [ ] I have run `brm validate` locally to verify the module files.
- [x] I have run deployment tests locally to ensure the module is deployable.

## Pipeline references

| Pipeline |
| - |
| [avm.res.network.bastion-host](https://github.com/krbar/bicep-registry-modules/actions/workflows/avm.res.network.bastion-host.yml) |

### CODE REVIEW

**Comment by matebarabas** [comment]
@krbar, please don't forget to uncomment the related entry in CODEOWNERS file as part of this PR. Thanks!

File: avm/res/network/bastion-host/tests/e2e/custompip/main.test.bicep [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name, location)}-diagnosticDependencies'
+  params: {
+    storageAccountName: 'dep${namePrefix}diasa${serviceShort}03'
+    logAnalyticsWorkspaceName: 'dep-${namePrefix}-law-${serviceShort}'
+    eventHubNamespaceEventHubName: 'dep-${namePrefix}-evh-${serviceShort}01'
+    eventHubNamespaceName: 'dep-${namePrefix}-evhns-${serviceShort}01'
+    location: location
+  }
+}
```
Comment (AlexanderSehr) [inline avm/res/network/bastion-host/tests/e2e/custompip/main.test.bicep:56]: Please remove as not required to show how to the use case works

File: avm/res/network/bastion-host/tests/e2e/waf-aligned/main.test.bicep [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+  location: location
+}
+
+module nestedDependencies 'dependencies.bicep' = {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name, location)}-nestedDependencies'
+  params: {
+    virtualNetworkName: 'dep-${namePrefix}-vnet-${serviceShort}'
+    publicIPName: 'dep-${namePrefix}-pip-${serviceShort}'
+    managedIdentityName: 'dep-${namePrefix}-msi-${serviceShort}'
```
Comment (AlexanderSehr) [inline avm/res/network/bastion-host/tests/e2e/waf-aligned/main.test.bicep:40]: I may be mistaken, but I think this dependency isn't actually used and could be removed

**Review by AlexanderSehr — CHANGES_REQUESTED** [review]
Note: The readme generation was updated in main

File: avm/res/network/bastion-host/tests/e2e/waf-aligned/main.test.bicep (hunk 2 above)
Comment (krbar) [reply avm/res/network/bastion-host/tests/e2e/waf-aligned/main.test.bicep]: Removed: [558a33e](https://github.com/Azure/bicep-registry-modules/pull/691/commits/558a33ef3184f1a7ed64e6b8783c34c14c990b0c)

File: avm/res/network/bastion-host/tests/e2e/custompip/main.test.bicep [hunk 3]
```diff
@@ -0,0 +1,109 @@
+targetScope = 'subscription'
+
+metadata name = 'Using custom public IP address'
```
[the remark this answers is not in the document]
Comment (krbar) [reply avm/res/network/bastion-host/tests/e2e/custompip/main.test.bicep]: what do you think of this: https://github.com/Azure/bicep-registry-modules/pull/691/commits/6a225d732b469243593e4407231606b4522ee80d ? Any suggestions?

**Comment by krbar** [comment]
> @krbar, please don't forget to uncomment the related entry in CODEOWNERS file as part of this PR. Thanks!

thanks, done

File: avm/res/network/bastion-host/tests/e2e/custompip/main.test.bicep (hunk 1 above)
Comment (AlexanderSehr) [reply avm/res/network/bastion-host/tests/e2e/custompip/main.test.bicep:56]: I missed that part. Thanks for clarifying :)

**Review by AlexanderSehr — APPROVED** [review]

**Pull request merged**